specs/engine.spec/src/main/resources/META-INF/zilla/core.idl

```diff
@@ -100,4 +100,31 @@ scope core
             octets extension;
         }
     }
+
+    scope event
+    {
+        enum Level
+        {
+            INFO,
+            WARNING,
+            ERROR
+        }
```

Comment:
I think this is a useful abstraction as INFO, WARNING, ERROR are quite generic and we can definitely map these to otlp (TRACE, DEBUG, INFO, WARN, ERROR, FATAL) & syslog (debug, info, notice, warning, err, crit, alert, emerg) but also expect to match these to any logging framework out there we may want to include in the future.

I think if we make this decision at the recording time, we can avoid the code duplication and potential inconsistency where we'd have to bake into each exporter the same mapping where we associate the severity levels for each event type.

Of course it's a tradeoff, so we can save the effort here and decide not to capture this information at recording time and implement the mapping in each of the exporters in the future. Pls LMK.

Reply:
I guess the point I'm making is that the intention of having a level at recording time is to control the verbosity and indicate severity, but that is not relevant at recording time for us because we record everything unconditionally.

We also have the trace level log implicit in the ring buffers (potentially augmented with events as discussed today), exposed via dump command to pcap.

So then what remains is to indicate what to export, which has different meaning per exporter concept, and we could always list the set of events we need to export explicitly. I suspect this will be further enhanced to define some groups for short hand, but the naming of those groups as they map to log levels is specific to the exporter, and we have no need to attempt a lowest common denominator approach at recording time.

Let's keep `Level` out of the events for now please.

Reply:
removed